java-frontend/src/main/java/org/sonar/java/resolve/BytecodeCompleter.java

```diff
@@ -83,6 +89,35 @@ public void complete(JavaSymbol symbol) {
     }
   }
 
+  @CheckForNull
+  public Object constantValue(Symbol owner, String constantName) {
```

Comment:
I think this method deserves quite some documentation to mention the fact that it is intended to be called from semantic resolution from source and that is why we need to read class bytes.

Reply:
Added some documentation in f7ea69796727befbad0ae1ec1e09982ec507b652